HW2/P2/parallel_vector.pyx

```diff
@@ -81,11 +82,28 @@ cpdef move_data_fine_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
+               # Get smaller and bigger index
+               small = min(src[idx], dest[idx])
```

Comment:
For the fine grained case, not sure if it is necessary to find the min/max and and check then lock separately, given that it is not partitioned. I simply acquire and release locks at the individual indices and it worked fine.